LinkedList/LinkedList/DoublyList.h

```diff
@@ -0,0 +1,429 @@
+		FNode* Previuos = nullptr;
+	};
+	FNode* Head = nullptr;
+	FNode* Tail = nullptr;
+
+public:
+	TDLList() {}
+
+	~TDLList() 
+	{
+		Clear();
+	}
+
+	//-----------------------------------CopyConstructor
+	TDLList(const TDLList<Type>& OtherList)
+	{
+		Clear();
+		FNode* CurrentNode = OtherList.Head;
+
+		while(CurrentNode)
+		{
+			AddTail(CurrentNode->Data);
+			CurrentNode = CurrentNode->Next;
+		}
+	}
+
+	//------------------------------------Iterator
+	class FIterator
+	{
+	private:
+		FNode* Current;
+	public:
+		FIterator(FNode* NewCurrent)
+		{
+			Current = NewCurrent;
+		}
+
+		FIterator& operator++()
+		{
+			Current = Current->Next;
+			return *this;
+		}
+
+		Type& operator*()
+		{
+			return Current->Data;
+		}
+
+		const Type& operator*() const
+		{
+			return Current->Data;
+		}
+
+		bool operator!=(const FIterator& Other)
+		{
+			return Current != Other.Current;
+		}
+
+		bool operator!=(const FIterator& Other) const
+		{
+			return Current != Other.Current;
+		}
+	};
+
+	FIterator begin()
+	{
+		return FIterator(Head);
+	}
+
+	FIterator end()
+	{
+		return FIterator(nullptr);
+	}
+	
+
+	//---------------------------------Access
+	Type& operator[](const int Index)
+	{
+		FNode* Current;
+		if (Index < Size / 2)
+		{
+			Current = Head;
+			for (int i = 0; i < Index / 2; i++)
+			{
+				Current = Current->Next;
+			}
+			return Current->Data;
+		}
+		else
+		{
+			Current = Tail;
+			for (int i = Size - 1; i < Index / 2; i--)
+			{
+				Current = Current->Previuos;
+			}
+			return Current->Data;
+		}
+	}
+
+	const Type& operator[](const int Index) const
+	{
+		FNode* Current;
+		if (Index < Size / 2)
+		{
+			Current = Head;
+			for (int i = 0; i < Index / 2; i++)
+			{
+				Current = Current->Next;
+			}
+			return Current->Data;
+		}
+		else
+		{
+			Current = Tail;
+			for (int i = Size - 1; i < Index / 2; i--)
+			{
+				Current = Current->Previuos;
+			}
+			return Current->Data;
+		}
+	}
+
+	//-------------------------------------------------------
+	TDLList& operator=(const TDLList<Type>& OtherList)
+	{
+		Clear();
+		FNode* CurrentNode = OtherList.Head;
+
+		for (int i = 0; i < OtherList.Size; i++)
+		{
+			AddTail(CurrentNode->Data);
+			CurrentNode = CurrentNode->Next;
+		}
+		return *this;
+	}
+
+	//--------------------------------------------------------
+	Type& GetHead()
+	{
+		return Head->Data;
+	}
+
+	const Type& GetHead() const
+	{
+		return Head->Data;
+	}
+
+	//---------------------------------------------------------
+	Type& GetTail()
+	{
+		return Tail->Data;
+	}
+
+	const Type& GetTail() const
+	{
+		return Tail->Data;
+	}
+
+	//---------------------------------Size
+	bool IsEmpty() const
+	{
+		return Size == 0;
+	}
+
+	//------------------------------------------------
+	int GetSize() const
+	{
+		return Size;
+	}
+
+	//-----------------------------------Modifiers
+	void AddHead(const Type& InHead)
+	{
+		FNode* NewHead = new FNode;
+		NewHead->Data = InHead;
+
+		if (Head == nullptr)
```

Comment:
Remember that you could've switched this around and either changed the inside to say `!Head` or switch the if and else contents and use `Head` here